src/circuits/joinsplit.tcc

```diff
@@ -94,7 +99,7 @@ class joinsplit_gadget : libsnark::gadget<FieldT> {
                 // The root is represented on a single field element
                 // Each nullifier, and each commitment are in {0,1}^256 and thus take 2 field elements to be represented,
                 // while value_pub_in, and value_pub_out are in {0,1}^64, and thus take a single field element to be represented
-                int nb_inputs = 1 + (2 * (NumInputs + NumOutputs)) + 1 + 1;
+                size_t nb_inputs = (2 * (NumInputs + NumOutputs)) + 1 + 1;
```

Comment:
Why did you remove the first `+1` @dtebbs ?

The number of primary inputs should be:
`1 (the merkle root =  a single field element) + 2 *(NumInputs + NumOutputs) (Each input nullifier gets packed into 2 field elements, likewise for each output commitment) + 1 (v_pub_in) + 1 (v_pub_out)`, right?

Reply:
Oh yes.  I forgot that value is also used to setup the protoboard, not just for the packed inputs length.  I'll fix it.

Reply:
(Sorry about that ... pushed another commit to fix this.  Feel free to squash or replace my commits.)

Reply:
Thanks